UnityProject/Assets/Scripts/Health/RespiratoryHealth/RespiratorySystem.cs

```diff
@@ -268,6 +370,146 @@ private void HandleBreathingPlasma(float plasmaAmount)
 				string.Format(plasmaHighOthersMessages.PickRandom(), gameObject.ExpensiveName(), theirPronoun)
 			);
 		}
+		return plasmaConsumed;
+	}
+
+	private bool HandleBreathingCarbonDioxide(GasMix gasMix)
+	{
+		float carbonPressure = gasMix.GetPressure(Gas.CarbonDioxide);
+		// there is a little carbon dioxide in the air.
+		if (carbonPressure <= CARBON_DIOXIDE_SAFE_MAX && carbonPressure > CARBON_DIOXIDE_WARNING_LEVEL)
+		{
+			if (DMMath.Prob(90))
+			{
+				return true;
+			}
+
+			// 10% chances of message
+			var theirPronoun = gameObject.Player() != null
+				? gameObject.Player().Script.characterSettings.TheirPronoun()
+				: "its";
+			Chat.AddActionMsgToChat(
+				gameObject,
+				CO2LowYouMessages.PickRandom(),
+				string.Format(
+					CO2LowOthersMessages.PickRandom(),
+					gameObject.ExpensiveName(),
+					string.Format(plasmaLowOthersMessages.PickRandom(), gameObject.ExpensiveName(), theirPronoun))
+			);
+		}
+		// enough carbon dioxide to start suffocating you!
+		else if (carbonPressure > CARBON_DIOXIDE_SAFE_MAX)
+		{
+			if (CarbonSuffocationTimer >= 0)
+				CarbonSuffocationTimer++;
+			if (CarbonSuffocationTimer >= 3 && CarbonSuffocationTimer < 7)
+			{
+				IsSuffocating = true;
+				bloodSystem.OxygenDamage += 3;
+
+				if (DMMath.Prob(90))
+				{
+					return true;
+				}
+
+				// 10% chances of message
+				var theirPronoun = gameObject.Player() != null
+					? gameObject.Player().Script.characterSettings.TheirPronoun()
+					: "its";
+				Chat.AddActionMsgToChat(
+					gameObject,
+					CO2MedYouMessages.PickRandom(),
+					string.Format(CO2MedOthersMessages.PickRandom(), gameObject.ExpensiveName(), theirPronoun)
+				);
+			}
+			else if (CarbonSuffocationTimer >= 7)
+			{
+				IsSuffocating = true;
+				bloodSystem.OxygenDamage += 8;
+
+				if (DMMath.Prob(90))
+				{
+					return true;
+				}
+
+				// 10% chances of message
+				var theirPronoun = gameObject.Player() != null
+					? gameObject.Player().Script.characterSettings.TheirPronoun()
+					: "its";
+				Chat.AddActionMsgToChat(
+					gameObject,
+					CO2HighYouMessages.PickRandom(),
+					string.Format(CO2HighOthersMessages.PickRandom(), gameObject.ExpensiveName(), theirPronoun)
+				);
+				return true;
+			}
+		}
+
+		if (carbonPressure < CARBON_DIOXIDE_WARNING_LEVEL)
+		{
+			CarbonSuffocationTimer = 0;
+			return (carbonPressure > 0);
+		}
+		return false;
+	}
+	private bool HandleWearingGasMask(GasMix gasMix)
```

Comment:
what does the return value mean for this method?

Reply:
the return value for HandleWearingGasMask or the value for HandleBreathingCarbonDioxide?

for HandleWearingGasMask, the return value tells the breathing script whether or not to go through with running the other checks.
for HandleBreathingCarbonDioxide, it's a bool that's used in the rare case that _only_ carbon dioxide is breathed in, to tell the rest of the program that you did, in fact, breathe that cycle